ksqldb-parser/src/main/java/io/confluent/ksql/parser/DefaultKsqlParser.java

```diff
@@ -130,4 +142,14 @@ private static String getStatementString(final SingleStatementContext singleStat
         singleStatementContext.stop.getStopIndex()
     ));
   }
+
+  //checks if the error is a reserved keyword error by checking the message and offendingSymbol
+  private static boolean isKeywordError(final String message, final Object offendingSymbol) {
+    final String tokenName = ((CommonToken) offendingSymbol).getText().toLowerCase();
+
+    final Pattern keywordPattern = Pattern.compile("extraneous input.*expecting.*");
+    final Matcher m = keywordPattern.matcher(message);
+    return KsqlParserUtil.isReserved(tokenName) && m.find();
```

Comment:
nit: we should flip the order here. `m.find()` is probably significantly cheaper than going through the full blown parser rules
```suggestion
    return m.find() && KsqlParserUtil.isReserved(tokenName);
```